src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -300,6 +300,12 @@ namespace winrt::TerminalApp::implementation
         ShowSetAsDefaultInfoBar();
     }
 
+    Automation::Peers::AutomationPeer TerminalPage::OnCreateAutomationPeer()
+    {
+        _autoPeer = Automation::Peers::FrameworkElementAutomationPeer(*this);
```

Comment:
Q: is it safe to cache this? Will it create a reference cycle that results in TerminalPage never deallocating?

Should we do this instead? (excerpt from CommandPalette)

```
        if (auto automationPeer{ Automation::Peers::FrameworkElementAutomationPeer::FromElement(_searchBox()) })

```

Reply:
also, this seems dangerous! does Page not have its own special type of automation peer?

Reply:
> is it safe to cache this?

We do this in `TermControl` so it should be safe. At least in our impl, we store a weak reference to the owner. I could implement a `TerminalPageAutomationPeer` that's based off of the `FrameworkElementAutomationPeer` though, if we want to be sure.

> Should we do this instead? (excerpt from CommandPalette)

I tried doing that before and `FromElement` can't find an automation peer (surprisingly)!

This actually just led me down a rabbit hole: should we be concerned that `FromElement()` didn't work? Looks like we shouldn't, actually! Check this out: https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.frameworkelementautomationpeer.fromelement?view=winrt-22621#remarks

Apparently, `FromElement()` relies on `OnCreateAutomationPeer`! So defining `OnCreateAutomationPeer` helps that out too. 

> does Page not have its own special type of automation peer?

Nope! "Control" doesn't even have it's own auto peer. It's standard to fall back to `FrameworkElementAutomationPeer` because it includes a bunch of basic behavior. See https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.frameworkelementautomationpeer?view=winrt-22621#remarks

Reply:
Let's not implement our own. I _am_ worried about the reference cycle, however. FrameworkElementAutomationPeer does keep a strong reference to its owner, as far as I can tell.

I am somewhat generally worried that we should _never_ cache these, actually.

[Docs say...](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.oncreateautomationpeer?view=winrt-22621)

> We recommend that the OnCreateAutomationPeer implementation should do nothing more than initialize a new instance of your custom automation peer, passing the calling control as owner, and return that instance. Do not attempt additional logic in this method. In particular, any logic that could potentially lead to destruction of the [AutomationPeer](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.automationpeer?view=winrt-22621) within the same call may result in unexpected runtime behavior.

I believe XAML caches it internally in `UIElement::m_tpAP`.

Reply:
Thanks for digging on the others :)